cpp/src/parquet/encoding.h

```diff
@@ -24,6 +24,7 @@
 
 #include "arrow/util/spaced.h"
 
+#include "arrow/type.h"
```

Comment:
we are already forward declaring elements below,  please do that instead of include type.

Reply:
I had to add this include because of the `EncodingTraits<ByteArrayType>::memory_limit` static member I have added. I believe constexpr values can't be forward declared or static members can't rely on forward declared symbols, smth like that.

What's your suggestion on this case? Remove the constexpr-ness from the symbols to be able to forward declare it, duplicate the definition (e.g memory_limit = std::numeric... on this file as well, might create inconsistencies?) or keep the include?

Reply:
Please check my comment in encoding.cc. I am not sure if that can help remove this header as well as `memory_limit` defined below.

Reply:
I believe I have addressed this, could you please take a look and confirm?